src/workshed.ts

```diff
@@ -0,0 +1,390 @@
+    }
+    if (handlingChoice()) visitUrl("main.php");
+  }
+);
+
+const pizzaCube = new GarboWorkshed(
+  $item`diabolic pizza cube`,
+  () => {
+    return dietCompleted;
+  },
+  () => {
+    return;
+  }
+);
+
+const asdonMartin = new GarboWorkshed(
+  $item`Asdon Martin keyfob`,
+  () => {
+    return haveEffect($effect`Driving Observantly`) >= estimatedTurns();
+  },
+  () => {
+    const currentTurns = haveEffect($effect`Driving Observantly`);
+    const requiredTurns = estimatedTurns() - currentTurns;
+    const currentFuel = getFuel();
+    const requiredFuel = Math.max(Math.ceil(requiredTurns / 30) * 37 - currentFuel, 0);
+    if (requiredFuel > 10000) {
+      throw new Error(`Wanted to fill ${requiredFuel} fuel in our Asdon, probably a bug!`);
+    }
+    fillTo(requiredFuel);
+    for (let x = 0; x < Math.ceil(requiredTurns / 30); x++) {
+      cliExecute("asdonmartin drive observantly");
+    }
+  }
+);
+
+const mayoClinic = new GarboWorkshed(
+  $item`portable Mayo Clinic`,
+  () => {
+    return dietCompleted;
+  },
+  () => {
+    return;
+  }
+);
+
+const DNALab = new GarboWorkshed(
+  $item`Little Geneticist DNA-Splicing Lab`,
+  () => {
+    // This will likely always return true or false for now, depending on the start state of garbo
+    // Since we don't actually support using the syringe in combat at this time, the counter will never change
+    return get("_dnaPotionsMade") >= 3;
+  },
+  () => {
+    // Just grab whatever tonics for now, since we don't actually have support for DNA
+    const page = visitUrl("campground.php?action=workshed");
+    if (page.includes("Human-")) {
+      visitUrl("campground.php?action=dnapotion");
+    }
+  }
+);
+
+const snowMachine = new GarboWorkshed(
+  $item`snow machine`,
+  () => {
+    // Assume garden has already been harvested(?)
+    return true;
+  },
+  () => {
+    return;
+  }
+);
+
+const spinningWheel = new GarboWorkshed(
+  $item`spinning wheel`,
+  () => {
+    return get("_spinningWheel");
+  },
+  () => {
+    // We simply assume you will not gain a level while garboing, since we do not do powerlevellings
+    // So we will just use the spinning wheel immediately
+    visitUrl("campground.php?action=spinningwheel");
+  }
+);
+
+const autoAnvil = new GarboWorkshed(
+  $item`warbear auto-anvil`,
+  () => {
+    // Garbo does not support meatsmithing, so assume we are done if the user is running garbo
+    return true;
+  },
+  () => {
+    return;
+  }
+);
+
+const chemistryLab = new GarboWorkshed(
+  $item`warbear chemistry lab`,
+  () => {
+    return potionSetupCompleted;
+  },
+  () => {
+    return;
+  }
+);
+
+const highEfficiencyStill = new GarboWorkshed(
+  $item`warbear high-efficiency still`,
+  () => {
+    return dietCompleted;
+  },
+  () => {
+    return;
+  }
+);
+
+const inductionOven = new GarboWorkshed(
+  $item`warbear induction oven`,
+  () => {
+    return dietCompleted;
+  },
+  () => {
+    return;
+  }
+);
+
+const drillPress = new GarboWorkshed(
```

Comment:
For all of these worksheds that do nothing, it would be better to have a function generate them:

```ts
const drillPress = noopWorkshed($item`warbear jackhammer drill press`);
```

Reply:
I used default methods instead, which I think looks nicer?